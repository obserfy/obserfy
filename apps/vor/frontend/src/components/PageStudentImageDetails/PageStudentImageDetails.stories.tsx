import { FC } from "react"
import PageStudentImageDetails from "./PageStudentImageDetails"

export default {
  title: "Core/PageStudentImageDetails",
  component: PageStudentImageDetails,
  parameters: {
    componentSubtitle: "Just a simple PageStudentImageDetails",
  },
}

export const Basic: FC = () => (
  <PageStudentImageDetails imageId="asdfasdf" studentId="asdfasdf" />
)
