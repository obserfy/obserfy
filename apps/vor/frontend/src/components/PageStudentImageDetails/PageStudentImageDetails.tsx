import { FC, useState } from "react"
import { Box, Button, Flex, Image } from "theme-ui"
import { t, Trans } from "@lingui/macro"
import BackNavigation from "../BackNavigation/BackNavigation"
import { STUDENT_GALLERY_URL } from "../../routes"
import useGetImage from "../../hooks/api/useGetImage"
import Typography from "../Typography/Typography"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import { usePatchStudentApi } from "../../hooks/api/students/usePatchStudentApi"
import AlertDialog from "../AlertDialog/AlertDialog"
import useDeleteImage from "../../hooks/api/useDeleteImage"
import { navigate } from "../Link/Link"
import dayjs from "../../dayjs"

interface Props {
  studentId: string
  imageId: string
}
export const PageStudentImageDetails: FC<Props> = ({ studentId, imageId }) => {
  const student = useGetStudent(studentId)
  const image = useGetImage(imageId)
  const [showDeleteDialog, setShowDeleteDialog] = useState(false)
  const [showSetProfileDialog, setShowSetProfileDialog] = useState(false)
  const deleteImage = useDeleteImage(studentId, imageId)
  const updateStudentImage = usePatchStudentApi(studentId)

  return (
    <>
      <Box sx={{ maxWidth: "maxWidth.md" }} margin="auto">
        <BackNavigation
          to={STUDENT_GALLERY_URL(studentId)}
          text={t`Image gallery`}
        />
        <Typography.H5 m={3}>{student.data?.name}</Typography.H5>
        <Box px={[0, 3]}>
          <Image src={image.data?.originalUrl} sx={{ width: "100%" }} />
        </Box>
        {!image.isLoading && (
          <Flex sx={{ alignItems: "center" }} mx={3} my={2}>
            <Typography.Body sx={{ fontSize: 1 }}>
              <Trans>Posted on</Trans>{" "}
              {dayjs(image.data?.createdAt).format("ddd, D MMM 'YY")}
            </Typography.Body>
            <Flex ml="auto">
              <Button
                variant="primary"
                mr={2}
                onClick={() => setShowSetProfileDialog(true)}
              >
                <Trans>Set as Profile</Trans>
              </Button>
              <Button
                variant="outline"
                color="danger"
                onClick={() => setShowDeleteDialog(true)}
              >
                <Trans>Delete</Trans>
              </Button>
            </Flex>
          </Flex>
        )}
      </Box>
      {showDeleteDialog && (
        <AlertDialog
          title={t`Delete image?`}
          positiveText={t`Delete`}
          body={t`Are you sure you want to delete this image?`}
          negativeText={t`Cancel`}
          onNegativeClick={() => setShowDeleteDialog(false)}
          loading={deleteImage.isLoading}
          onPositiveClick={async () => {
            try {
              await deleteImage.mutateAsync()
              await navigate(STUDENT_GALLERY_URL(studentId))
            } catch (e) {
              Sentry.captureException(e)
            }
          }}
        />
      )}

      {showSetProfileDialog && (
        <AlertDialog
          title={t`Set as profile?`}
          positiveText={t`Yes`}
          body={t`
            Are you sure you want to set this image as profile picture?
          `}
          negativeText={t`Cancel`}
          onNegativeClick={() => setShowSetProfileDialog(false)}
          loading={updateStudentImage.isLoading}
          onPositiveClick={async () => {
            try {
              await updateStudentImage.mutateAsync({
                profileImageId: imageId,
              })
              setShowSetProfileDialog(false)
            } catch (e) {
              Sentry.captureException(e)
            }
          }}
        />
      )}
    </>
  )
}

export default PageStudentImageDetails
