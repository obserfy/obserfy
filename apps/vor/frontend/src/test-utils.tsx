/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
