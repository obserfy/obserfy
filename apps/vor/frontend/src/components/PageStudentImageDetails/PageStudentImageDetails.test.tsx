import { render } from "../../test-utils"
import { Basic } from "./PageStudentImageDetails.stories"

describe("PageStudentImageDetails", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
