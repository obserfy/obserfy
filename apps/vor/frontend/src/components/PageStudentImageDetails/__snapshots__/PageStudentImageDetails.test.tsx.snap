// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageStudentImageDetails should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: auto;
  max-width: 700px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0;
  padding-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-6 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/students/gallery?studentId=asdfasdf"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            Image gallery
          </p>
        </div>
      </a>
      <h5
        class="emotion-5"
      />
      <div
        class="emotion-6"
      >
        <img
          class="emotion-7"
        />
      </div>
    </div>
  </div>
</div>
`;
